src/http_client_async.zig

```diff
@@ -363,25 +364,10 @@ fn NewHTTPContext(comptime ssl: bool) type {
             if (!comptime ssl) {
                 unreachable;
             }
-            var opts: uws.us_bun_socket_context_options_t = .{
-                .request_cert = 1,
-                .reject_unauthorized = 0,
-            };
-            if (client.custom_tls_props.passphrase) |passphrase| {
-                opts.passphrase = passphrase;
-            }
-            if (client.custom_tls_props.ca) |ca| {
-                opts.ca = ca.ptr;
-                opts.ca_count = 1;
-            }
-            if (client.custom_tls_props.cert) |cert| {
-                opts.cert = cert.ptr;
-                opts.cert_count = 1;
-            }
-            if (client.custom_tls_props.key) |key| {
-                opts.key = key.ptr;
-                opts.key_count = 1;
-            }
+
+            var opts = client.tls_props.?.asUSockets();
+            opts.request_cert = 1;
+            opts.reject_unauthorized = 1;
```

Comment:
`reject_unauthorized` should be 0 here we handle this error in another place checking the certificate